### PR TITLE
Add GTM tracking on new search

### PR DESCRIPTION
**Pull request by jondewijones**
### What

- add GTM tracking for each search result
- add data to datalayer
- added new helper, also been added to new renderer library https://github.com/ONSdigital/dp-renderer/pull/14
- update main.tmpl to prefix type class on body element with .page-type-- to prevent page type class from causing unwanted changes e.g now uses .page-type--search rather than .search which causes issues, also added to new renderer library

### Who can review

Anyone

### CODE REVIEW

File: assets/templates/search/results.tmpl [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+{{ $currentPage := .Data.Pagination.CurrentPage }}
+{{ $itemsPerPage := .Data.Pagination.Limit }}
 <div id="results" class="results">    
     {{ if ne .Data.Response.Count 0 }}
         <div class="search-results ">
             <ul class="list--neutral flush">
-                {{ range .Data.Response.Items }}
+                {{ range $i, $item := .Data.Response.Items }}
+                    {{ $currentPosition := add $i 1 }}
+                    {{ $totalSearchPosition := multiply (subtract $currentPage 1) $itemsPerPage }}
```
Comment (dandwal) [inline assets/templates/search/results.tmpl:10]: if $currentPage starts of as `0`
`{{ $totalSearchPosition := multiply (subtract $currentPage 1) $itemsPerPage }}`
Would result in $totalSearchPosition being `-1*$itemsPerPage` Beign negative I assume .Data.Pagination.CurrentPage is always `>1` ?

This line probably should be bumped to be above line 8 somewhere (maybe even just after line 3, as it isn't using and variables declared within the range statement but is being redeclared in the rage statement numerous times and each time yielding the same value.

File: render/helpers_test.go [hunk 2]
```diff
@@ -24,6 +24,12 @@ func TestSubtract(t *testing.T) {
 	})
 }
 
+func TestMultiply(t *testing.T) {
+	Convey("substract should return expected value", t, func() {
```
Comment (dandwal) [inline render/helpers_test.go:28]: C/P?

File: render/helpers_test.go [hunk 3]
```diff
@@ -238,6 +244,14 @@ func TestDatePeriodFormat(t *testing.T) {
 
 }
 
+func TestDateFormatYYYYMMDDNoSlash(t *testing.T) {
+	Convey("Date format returns human readable string", t, func() {
```
Comment (dandwal) [inline render/helpers_test.go:248]: I maybe wouldn't opt to use the word 'readable' here 😅

**Review by dandwal — COMMENTED** [review]
Haven't checked but have assumed we don't have any classes named our page types in GA/GTM or in the actual CSS and JS.

File: render/helpers_test.go (hunk 2 above)
Comment (jondewijones) [reply render/helpers_test.go:28]: 👍

File: render/helpers_test.go (hunk 3 above)
Comment (jondewijones) [reply render/helpers_test.go:248]: 👍

File: assets/templates/search/results.tmpl (hunk 1 above)
Comment (jondewijones) [reply assets/templates/search/results.tmpl]: Yup `currentPage` is always (or should always, and if not we probably have bigger issues)`>0`. 

Good spot. Think it was originally using something in the loop but isn't any more. Moved code out of loop.

**Review by dandwal — APPROVED** [review]
LGTM

**Pull request merged**